File: day01/day01.py
#!/usr/bin/python3
from math import prod
from itertools import combinations

def readInput(filename):
    file = open(filename, "r")
    data = [int(value) for value in file.readlines()]

    return data


def solve(elements):
    for combination in combinations(data, elements):
        if sum(combination) == 2020:
            return prod(combination)


data = readInput("input.txt")
print("[Part 1]: {:d}".format(solve(2)))
print("[Part 2]: {:d}".format(solve(3)))

File: day01/day01.php
<?php

class day01 {
    private array $numbers;

    public function readInput(string $filename) : void {
        $input = file_get_contents(filename: $filename);
        $input = array_map("intval", explode("\n", $input));

        $this->numbers = $input;
    }

    public function getPair(int $sum) : int {
        foreach ($this->numbers as $number) {
            $complement = $sum - $number;
            if (in_array($complement, $this->numbers)) {
                return $complement * $number;
            }
        }
        
        return -1;
    }

    public function getTriplet(int $sum) : int {
        foreach ($this->numbers as $first) {
            foreach ($this->numbers as $second) {
                foreach ($this->numbers as $third) {
                    if ($first + $second + $third === $sum) {
                        return $first * $second * $third;
                    }
                }
            }
        }

        return -1;
    }
}

$solution = new day01();
$solution->readInput(filename: "input.txt");

$p1 = $solution->getPair(sum: 2020);
echo("Part one: {$p1}\n");

$p2 = $solution->getTriplet(sum: 2020);
echo("Part two: {$p2}\n");